paperqa/types.py

```diff
@@ -594,6 +600,8 @@ def populate_bibtex_key_citation(  # noqa: PLR0912
     @model_validator(mode="before")
     @classmethod
     def validate_all_fields(cls, data: dict[str, Any]) -> dict[str, Any]:
+        if isinstance(data, Doc):  # type: ignore[unreachable]
```

Comment:
I think we should type hint `data: dict[str, Any] | Doc`, then it will resolve this `unreachable`

Also, can you add a unit test proving we don't manipulate an input `dict`?

We shouldn't be manipulating the input `dict` during validations